### PR TITLE
FAI-8191 - Add global source option to compress state

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

I made this optional instead of enforced because this is a public cdk and usually people want uncompressed state for easier inspection/debugging/modifying.

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

**Review by ypc-faros — APPROVED** [review]

File: faros-airbyte-cdk/src/protocol.ts [hunk 1]
```diff
@@ -50,7 +50,10 @@ export enum AirbyteTraceFailureType {
 }
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface AirbyteConfig {}
+export interface AirbyteConfig {
+  compress_state?: boolean;
```
Comment (tovbinm) [inline faros-airbyte-cdk/src/protocol.ts:54]: where will you be setting it to `true`?

Comment (cjwooo) [reply faros-airbyte-cdk/src/protocol.ts:54]: The user sets it to true, e.g. with cli: `--src.compress_state true`

Comment (tovbinm) [reply faros-airbyte-cdk/src/protocol.ts:54]: This is a breaking change for our connectors. Please make sure to reflect it in the airbyte-connectors-configs

Comment (cjwooo) [reply faros-airbyte-cdk/src/protocol.ts:54]: This is not a breaking change. The param doesn't need to be set, and you can switch between compressed and uncompressed whenever you want.

Comment (tovbinm) [reply faros-airbyte-cdk/src/protocol.ts:54]: currently we compress the state by default, and a lot of our connectors have a very large state which they should be compressing.

File: faros-airbyte-cdk/src/sources/source-base.ts [hunk 2]
```diff
@@ -131,8 +132,15 @@ export abstract class AirbyteSourceBase<
           configuredStream,
           connectorState
         );
+
         for await (const message of generator) {
-          yield message;
+          if (isStateMessage(message) && config.compress_state) {
+            yield new AirbyteStateMessage({
+              data: State.compress(message.state.data),
```
Comment (tovbinm) [inline faros-airbyte-cdk/src/sources/source-base.ts:139]: dont forget the `message.state.sourceStatus`

**Pull request merged**